### PR TITLE
Fix issue where snapshot conflicts weren't being handled

### PR DESCRIPTION
**Pull request by mikemountain**
If there is already information in sessions_pending_daily_snapshot then the larger part of the update_sessions_pending_daily_snapshot() doesn't seem to work properly:
Initial insert to empty table:
```
boundary=# select * from update_sessions_pending_daily_snapshot();
 snapshot_date | sessions_pending_count 
---------------+------------------------
 2023-07-06    |                      0
 2023-07-05    |                      0
 2023-07-04    |                      0
 2023-07-03    |                      1
 2023-07-02    |                      1
(5 rows)
```
running it again to see nothing happens (hits first if case)
```
boundary=# select * from update_sessions_pending_daily_snapshot();
 snapshot_date | sessions_pending_count 
---------------+------------------------
(0 rows)
```
delete yesterday's record to have a day be "missing"
```
boundary=# delete from sessions_pending_daily_snapshot where snapshot_date = '2023-07-06';
DELETE 1
```
running it again:
```
boundary=# select * from update_sessions_pending_daily_snapshot();
ERROR:  duplicate key value violates unique constraint "sessions_pending_daily_snapshot_pkey"
DETAIL:  Key (snapshot_date)=(2023-07-02) already exists.
CONTEXT:  PL/pgSQL function update_sessions_pending_daily_snapshot() line 21 at RETURN QUERY
```

### CODE REVIEW

**Review by johanbrandhorst — APPROVED** [review]
Would it be possible for us to test this?

**Review by johanbrandhorst — APPROVED** [review]

File: internal/db/schema/migrations/oss/postgres/74/01_hcp_update_sessions_function.up.sql [hunk 1]
```diff
@@ -0,0 +1,75 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: MPL-2.0
+
+begin;
+
+-- replaces function from 70/01_hcp_billing_daily.up.sql
+drop function update_sessions_pending_daily_snapshot;
```
Comment (mgaffney) [inline internal/db/schema/migrations/oss/postgres/74/01_hcp_update_sessions_function.up.sql:7]:
```suggestion
  drop function update_sessions_pending_daily_snapshot;
```

File: internal/db/schema/migrations/oss/postgres/74/01_hcp_update_sessions_function.up.sql [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+create function update_sessions_pending_daily_snapshot()
```
Comment (mgaffney) [inline internal/db/schema/migrations/oss/postgres/74/01_hcp_update_sessions_function.up.sql:9]:
```suggestion
  create function update_sessions_pending_daily_snapshot()
```

**Review by mgaffney — APPROVED** [review]
Looks great. I left some minor indentation suggestions.

**Pull request merged**